katdal/sensordata.py

```diff
@@ -249,37 +376,49 @@ def dummy_sensor_data(name, value=None, dtype=np.float64, timestamp=0.0):
             value = ''
         elif np.issubdtype(dtype, np.bool):
             value = False
+        elif np.issubdtype(dtype, np.object):
+            value = None
     else:
         dtype = np.array(value).dtype
-    data = np.rec.fromarrays([[timestamp], [value], ['nominal']],
-                             dtype=[('timestamp', np.float64), ('value', dtype), ('status', '|S7')])
-    return SensorData(data, name)
+    data = np.array(zip([timestamp], [value]),
+                    dtype=[('timestamp', np.float64), ('value', dtype)])
+    return RecordSensorData(data, name)
```

Comment:
Indeed... To preserve consistency, it needs wrapping too, as all SensorData-stage operations should deal with wrapped objects. If the filler value is e.g. ```[]``` it will get the wrong dtype too, so this now uses my freshly minted ```infer_dtype``` function instead.

Thanks for picking that up. Fixed.